prov/util: enum-conversion warning in cuda_mem_monitor.c

**Describe the bug**
When building with `-Werror -Wenum-conversion` flags I get a build failure in the util provider cuda code.

**To Reproduce**
Compile with clang or ICX (intel's new compiler):
```
./configure --enable-only --enable-shm \
   --enable-cuda-dlopen --with-cuda=/usr/local/cuda \
   CFLAGS="-Werror -Wenum-conversion"
```
enable shm is to just speed up the process. 

**Expected behavior**
Compile library without issue

**Output**
```
prov/util/src/cuda_mem_monitor.c:57:53: error: implicit conversion from enumeration type 'CUresult' (aka 'enum cudaError_enum') to different enumeration type 'cudaError_t' (aka 'enum cudaError') [-Werror,-Wenum-conversion]
                ofi_cudaGetErrorName(ret), ofi_cudaGetErrorString(ret));
                                           ~~~~~~~~~~~~~~~~~~~~~~ ^~~
./include/rdma/providers/fi_log.h:97:39: note: expanded from macro 'FI_WARN'
        FI_LOG(prov, FI_LOG_WARN, subsystem, __VA_ARGS__)
                                             ^~~~~~~~~~~
./include/rdma/providers/fi_log.h:80:25: note: expanded from macro 'FI_LOG'
                                __func__, __LINE__, __VA_ARGS__);       \
                                                    ^~~~~~~~~~~
prov/util/src/cuda_mem_monitor.c:57:24: error: implicit conversion from enumeration type 'CUresult' (aka 'enum cudaError_enum') to different enumeration type 'cudaError_t' (aka 'enum cudaError') [-Werror,-Wenum-conversion]
                ofi_cudaGetErrorName(ret), ofi_cudaGetErrorString(ret));
                ~~~~~~~~~~~~~~~~~~~~ ^~~
./include/rdma/providers/fi_log.h:97:39: note: expanded from macro 'FI_WARN'
        FI_LOG(prov, FI_LOG_WARN, subsystem, __VA_ARGS__)
                                             ^~~~~~~~~~~
./include/rdma/providers/fi_log.h:80:25: note: expanded from macro 'FI_LOG'
                                __func__, __LINE__, __VA_ARGS__);       \
                                                    ^~~~~~~~~~~
prov/util/src/cuda_mem_monitor.c:96:54: error: implicit conversion from enumeration type 'CUresult' (aka 'enum cudaError_enum') to different enumeration type 'cudaError_t' (aka 'enum cudaError') [-Werror,-Wenum-conversion]
                        ofi_cudaGetErrorName(ret), ofi_cudaGetErrorString(ret));
                                                   ~~~~~~~~~~~~~~~~~~~~~~ ^~~
./include/rdma/providers/fi_log.h:97:39: note: expanded from macro 'FI_WARN'
        FI_LOG(prov, FI_LOG_WARN, subsystem, __VA_ARGS__)
                                             ^~~~~~~~~~~
./include/rdma/providers/fi_log.h:80:25: note: expanded from macro 'FI_LOG'
                                __func__, __LINE__, __VA_ARGS__);       \
                                                    ^~~~~~~~~~~
prov/util/src/cuda_mem_monitor.c:96:25: error: implicit conversion from enumeration type 'CUresult' (aka 'enum cudaError_enum') to different enumeration type 'cudaError_t' (aka 'enum cudaError') [-Werror,-Wenum-conversion]
                        ofi_cudaGetErrorName(ret), ofi_cudaGetErrorString(ret));
                        ~~~~~~~~~~~~~~~~~~~~ ^~~
./include/rdma/providers/fi_log.h:97:39: note: expanded from macro 'FI_WARN'
        FI_LOG(prov, FI_LOG_WARN, subsystem, __VA_ARGS__)
                                             ^~~~~~~~~~~
./include/rdma/providers/fi_log.h:80:25: note: expanded from macro 'FI_LOG'
                                __func__, __LINE__, __VA_ARGS__);       \
                                                    ^~~~~~~~~~~
4 errors generated.
make[1]: *** [Makefile:27828: prov/util/src/src_libfabric_la-cuda_mem_monitor.lo] Error 1
make[1]: *** Waiting for unfinished jobs....
prov/coll/src/coll_coll.c:1242:30: error: variable 'coll_op' set but not used [-Werror,-Wunused-but-set-variable]
        struct util_coll_operation *coll_op;
                                    ^
1 error generated.
make[1]: *** [Makefile:27884: prov/coll/src/src_libfabric_la-coll_coll.lo] Error 1
```

**Environment:**
Linux: `RHEL 8.7`
CLANG: `clang version 14.0.6 (Red Hat 14.0.6-1.module+el8.7.0+15816+ec020e8f)`
ICX: `Intel(R) oneAPI DPC++/C++ Compiler 2023.0.0 (2023.0.0.20221201)`

**Additional context**
`cudaGetErrorString()` and `cudaError_t` is part of the CUDA (runtime) API, while `cuGetErrorName()` and `CUresult` are from the CU (driver) API. 

Technically there is some overlap in values, but they are not exactly the same. 
Possible Options:
* (lazy) just typecast:
```diff
-                ofi_cudaGetErrorName(ret), ofi_cudaGetErrorString(ret));
+                ofi_cudaGetErrorName((cudaError_t)ret), ofi_cudaGetErrorString((cudaError_t)ret));
```
* add `ofi_cuGetError*()` versions of functions and call them here instead.

--- a/include/ofi_hmem.h
+++ b/include/ofi_hmem.h
@@ -58,6 +58,8 @@ cudaError_t ofi_cudaMemcpy(void* dst, const void* src, size_t count,
 			   enum cudaMemcpyKind kind);
 const char *ofi_cudaGetErrorName(cudaError_t error);
 const char *ofi_cudaGetErrorString(cudaError_t error);
+CUresult ofi_cuGetErrorName(CUresult error, const char** pStr);
+CUresult ofi_cuGetErrorString(CUresult error, const char** pStr);
 CUresult ofi_cuPointerGetAttribute(void *data, CUpointer_attribute attribute,
 				   CUdeviceptr ptr);
 cudaError_t ofi_cudaHostRegister(void *ptr, size_t size, unsigned int flags);

--- a/prov/util/src/cuda_mem_monitor.c
+++ b/prov/util/src/cuda_mem_monitor.c
@@ -40,6 +40,7 @@ static int cuda_mm_subscribe(struct ofi_mem_monitor *monitor, const void *addr,
 			     size_t len, union ofi_mr_hmem_info *hmem_info)
 {
 	CUresult ret;
+	const char *errname, *errstr;
 
 	ret = ofi_cuPointerGetAttribute(&hmem_info->cuda_id,
 					CU_POINTER_ATTRIBUTE_BUFFER_ID,
@@ -51,10 +52,12 @@ static int cuda_mm_subscribe(struct ofi_mem_monitor *monitor, const void *addr,
 		return FI_SUCCESS;
 	}
 
+	ofi_cuGetErrorName(ret, &errname);
+	ofi_cuGetErrorString(ret, &errstr);
 	FI_WARN(&core_prov, FI_LOG_MR,
 		"Failed to get CUDA buffer ID for buffer %p len %lu\n"
 		"cuPointerGetAttribute() failed: %s:%s\n", addr, len,
-		ofi_cudaGetErrorName(ret), ofi_cudaGetErrorString(ret));
+		errname, errstr);
 
 	return -FI_EFAULT;
 }
@@ -72,6 +75,7 @@ static bool cuda_mm_valid(struct ofi_mem_monitor *monitor,
 {
 	uint64_t id;
 	CUresult ret;
+	const char *errname, *errstr;
 
 	/* CUDA buffer IDs are associated for each CUDA monitor entry. If the
 	 * device pages backing the device virtual address change, a different
@@ -89,11 +93,13 @@ static bool cuda_mm_valid(struct ofi_mem_monitor *monitor,
 		       "CUDA buffer ID %lu invalid for buffer %p\n",
 		       entry->hmem_info.cuda_id, entry->info.iov.iov_base);
 	} else {
+		ofi_cuGetErrorName(ret, &errname);
+		ofi_cuGetErrorString(ret, &errstr);
 		FI_WARN(&core_prov, FI_LOG_MR,
 			"Failed to get CUDA buffer ID for buffer %p len %lu\n"
 			"cuPointerGetAttribute() failed: %s:%s\n",
 			entry->info.iov.iov_base, entry->info.iov.iov_len,
-			ofi_cudaGetErrorName(ret), ofi_cudaGetErrorString(ret));
+			errname, errstr);
 	}
 
 	return false;

